app/models/miq_request.rb

```diff
@@ -456,6 +458,23 @@ def self.create_request(values, requester, auto_approve = false)
 
     request.post_create(auto_approve)
   end
+  api_relay_class_method(:create_request, :create) do |values, requester, auto_approve|
+    [
+      find_source_id_from_values(values),
+      {
+        :options      => values,
+        :requester    => {"user_name" => requester.userid},
+        :auto_approve => auto_approve
+      }
+    ]
+  end
+
+  def self.find_source_id_from_values(values)
+    MiqRequestMixin.get_option(:src_vm_id, nil, values) ||
+      MiqRequestMixin.get_option(:src_id, nil, values) ||
+      MiqRequestMixin.get_option(:src_ids, nil, values).try(:first)
```

Comment:
Maybe we need to divide and conquer? Splitting the src_ids by their regions and creating a request for each one?